Library/Homebrew/cmd/create.rb

```diff
@@ -107,17 +112,25 @@ def url=(url)
         /(.*?)[-_.]?#{path.version}/.match path.basename
         @name = $1
       end
-      @path = Formulary.path @name unless @name.nil?
-    else
-      @path = Formulary.path name
     end
+    update_path
     if @version
       @version = Version.new(@version)
     else
       @version = Pathname.new(url).version
     end
   end
 
+  def update_path
+    if @name
+      if @tap.nil?
+        @path = Formulary.core_path @name
+      else
+        @path = Formulary.tap_path @name, @tap
+      end
+    end
```

Comment:
I don't think we need to extend `Formulary`. All of this can be simplified to:

```ruby
return if @name.nil?
@path = Formulary.path("#{tap || "homebrew/core"}/#{name}")
```